### DISCUSSION TITLE
What is the best way to include this within a React application?

### DISCUSSION THREAD

**Discussion by farhad-ghayour** [SVG-Edit/svgedit#731] [Q&A]
Thank you for this awesome project. I was just experimenting with the React extension and love the module architecture of extensions. As I'd considering this project within a larger React context, what would be the best way to include it for development?

At first, I was thinking I'd 1) clone the entire project within the React project, 2) build it and 3) refer to the root JS/HTML? Something doesn't feel right here, as I'm not sure which is the root and this would not allow for rapid local development as it requires constantly building SVGEdit as well (for changes).

**Comment by jfhenon** [comment]
What about including the project as a dependency of your react project?

**Reply by farhad-ghayour** [reply]
Thanks for the response, @jfhenon. I'm assuming there is no `npm install` here; if so, what would be the root (that would include all of the functionaity) to include as a dependency?

**Comment by jfhenon** [comment]
There is a npm install!  (npm i -s svgedit)
After this, you can import svgedit into your code (``import Editor from 'svgedit'``)
You need to also have the CSS in your project (it's in the dist/Editor folder)
after you can create an instance of svgEdit  and attached to any div of your project ``const svgEditor = new Editor(myDiv)``

PS: The package is not updated as fast as the master branch but if you need a publish fr your project, I can do it.

**Reply by farhad-ghayour** [reply]
That would be amazing, @jfhenon! Just got it working with the `index.html` but the approach with `npm` would be even more awesome. Curious, what are the steps you're taking to make it as an exposable `npm` module? I ask incase I do it myself and want to always have the latest?

**Reply by vidjuheffex** [reply]
It seems it points to the wrong place, package.json has it pointing to `dist/Editor` where it actually live in `dist/editor/Editor`

That said, when I do the above approach I get `cmd should not be null` and nothing renders.

**Reply by mike2003** [reply]
> import svgedit into your code (import Editor from 'svgedit')

I have this "Cannot find module 'svgedit' or its corresponding type declarations."
What am I supposed to do?
